### PR TITLE
Implementing callers for SCAADC readouts

### PR DESCRIPTION
**Pull request by mexanick**
## Description

Caller wrapper for SCA ADC readout
Requires clarification on return documentation - see https://github.com/cms-gem-daq-project/ctp7_modules/issues/142
### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

Not yet tested, WIP
## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by bdorney** [comment]
Is this `WIP`? Review was requested but I see `In Progress` tag and `WIP` in title; please advise.

**Comment by mexanick** [comment]
Yes, review with suggestion on documenting the resulting array (from @ram1123) is expected. Of course, other comments are also welcome.

**Comment by mexanick** [comment]
@ram1123 please provide feedback

File: xhalcore/src/common/rpc_manager/sca.cc [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    STANDARD_CATCH;
+
+    if (rsp.get_key_exists("error")) {
+        printf("Caught an error: %s\n", (rsp.get_string("error")).c_str());
+        return 1;
+    }
+    const uint32_t size = count_1bits(ohMask);
+
+    if (rsp.get_key_exists("data")){
+        ASSERT(rsp.get_word_array_size("data") == size);
```
Comment (ram1123) [inline xhalcore/src/common/rpc_manager/sca.cc:24]: @mexanick : Are you sure that number of 1's in `ohMask` is always equal to 1's in `data`. Lets say if there is only one OH is unmasked its its **size** will be 1 but data might have more than 1 ones.

I don't have much idea so I might be wrong.

**Review by ram1123 — COMMENTED** [review]
After this small comments. From my side its fine.

File: xhalcore/src/common/rpc_manager/sca.cc (hunk 1 above)
Comment (jsturdy) [reply xhalcore/src/common/rpc_manager/sca.cc:24]: Reading a single sensor will only ever return one value per unmasked OH, so in this case it's "fine".
For L48, it's multiplied by 5, a magic number which presumably indicates the number of temperature sensors...
Likewise in L80, L112, and L144, the magic numbers 6, 3, and 14
I don't like this, but if it works, then it serves its purpose for now, and this code *won't* be migrated into the templated framework

**Review by jsturdy — APPROVED** [review]
If it works, I'll sign off, as this is *just* for the legacy "get-something-working-now" branch

**Review by ram1123 — APPROVED** [review]

**Pull request merged**